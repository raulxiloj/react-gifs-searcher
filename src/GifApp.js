import React, { useState } from 'react'
import { AddCategory } from './components/AddCategory';
import { GifCollection } from './components/GifCollection';

const GifApp = () => {

    let [categories, setCategories] = useState(["Bob esponja"]);

    //const handleAdd = () => {
    //    setCategories([...categories, 'Spiderman']);
    //    //setCategories(categories => [...categories, 'Spiderman']); 
    //};

    return (
        <>
            <h1 id="title">Giph App</h1>
            <AddCategory setCategories={setCategories} />
            <hr />
            <ol>
                {
                    categories.map(category =>
                        <GifCollection key={category} category={category} />
                    )
                }
            </ol>
        </>
    )
}

export default GifApp;
