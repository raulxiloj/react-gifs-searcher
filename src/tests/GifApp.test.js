import React from 'react';
import { shallow } from "enzyme"
import GifApp from "../GifApp"

describe('Tests for GifApp component', () => {

    test('It should render the component correctly', () => {
        const wrapper = shallow(<GifApp />)
        expect(wrapper).toMatchSnapshot();
    })

})
