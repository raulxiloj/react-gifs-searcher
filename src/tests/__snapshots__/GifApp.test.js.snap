// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for GifApp component It should render the component correctly 1`] = `
<Fragment>
  <h1
    id="title"
  >
    Giph App
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifCollection
      category="Bob esponja"
      key="Bob esponja"
    />
  </ol>
</Fragment>
`;
